pdns/dnsdistdist/test-dnsdistasync.cc

```diff
@@ -167,9 +167,9 @@ BOOST_AUTO_TEST_CASE(test_AddingExpiredEvent)
   // the expired event should be triggered almost immediately,
   // but we have seen many spurious failures on our CI,
   // likely because the box is overloaded, so sleep for up to
-  // 100 ms to be sure
+  // 400 ms to be sure
   for (size_t counter = 0; !holder->empty() && counter < 10; counter++) {
```

Comment:
Would it help to change the condition to something like this instead?
```suggestion
  for (size_t counter = 0; counter < 10; counter++) {
    if (holder->empty() && sender->errorRaised.load()) {
      break;
    }
    usleep(10000);
  }
```

Reply:
It should, will check tomorrow

Reply:
And it does. BTW. I'm also seeing this on FreeBSD once in a while and assuming the same fix will work on FreeBSD as well. If not, I'll revisit.